src/main/java/org/opentestsystem/ap/ims/client/GitClient.java

```diff
@@ -48,7 +49,21 @@
 import static java.lang.System.out;
 
 /**
- * A GIT client using JGit. Cloned repositories are expected to exist under the group in GitLab.
+ * A GIT client using JGit.
+ * <p>
+ * It is expected one will use {@link GitClientFactory} to instantiate an instance of this
+ * class.  The name of the repository to work with is given to the factory.  Once there
+ * is an instance of this class it can be used to interact with that repository.  The
+ * main point is an instance of htis class is not stateless.  If you need to work with two different
+ * repositories you need create two different instances of this class.
+ * </p>
+ * <p>
+ * The methods in this class are primarily Git commands.  The methods try to do one
+ * thing and one thing only, for the most part.  It is up to the user
+ * to use these commands in the order that works for them.  For example, one could switch
+ * to master branch, pull the latest, and then push. That would be three calls
+ * to this class, one for each of those commands.
+ * </p>
  */
 @Slf4j
 public class GitClient {
```

Comment:
Are you asking about this class specifically or is this more general toward "always code to interfaces"?  I ask because I removed most of the interfaces and decided to only introduce those if needed.  Most of the time it isn't needed as there is only ever one implementation.  

So far I have not seen a need for an interface in regards to testing.  Everything is mocked and there haven't been issues mocking classes.   There is potential for integration tests to need an actual implementation that differs from the production one.  But until that comes up I prefer not to introduce the clutter.

I originally had GitClient as an interface, and the implementation GitClientImpl and early on I had it called something like JGitClientImpl.  I backed off that and now the app feels lighter, less cluttered, and easier to understand.